src/cmd/services/m3aggregator/config/aggregator.go

```diff
@@ -689,7 +689,10 @@ func (c flushManagerConfiguration) NewFlushManagerOptions(
 		opts = opts.SetMaxJitterFn(maxJitterFn)
 	}
 	if c.NumWorkersPerCPU != 0 {
-		workerPoolSize := int(float64(runtime.NumCPU()) * c.NumWorkersPerCPU)
+		var workerPoolSize int
```

Comment:
super nit: Could you slightly simplify this to

```golang
var workerPoolSize = int(float64(runtime.NumCPU()) * c.NumWorkersPerCPU)
if workerPoolSize < 1 {
    // Don't allow the worker pool to be configured such that it does no work.
    workerPoolSize = 1
}
```

Alternatively,

```golang
var workerPoolSize = int(float64(runtime.NumCPU()) * c.NumWorkersPerCPU)
if workerPoolSize < 1 {
    return nil, fmt.Errorf("aggregator worker pool configured to size zero, check numWorkerPerCPU configuration")
}
```

Might be even better.

Reply:
Sure, changed to that. Added number of cpus to the error message as well to avoid any "What do you mean configured to 0, I have clearly configured it to 0.5" confusion.